### PR TITLE
feat: Custom ca cert support

### PR DESCRIPTION
**Pull request by jacobbmay**

### CODE REVIEW

File: bundles/uds-core-swf/uds-bundle.yaml [hunk 1]
```diff
@@ -619,10 +702,15 @@ packages:
               value:
                 - host: nexus-docker.${DOMAIN}
                   port: 5000
+            - path: deployment.additionalVolumes
+              value: *extra-volumes
+            - path: deployment.additionalVolumeMounts
+              value: *extra-volume-mounts
         uds-nexus-config:
+          values:
```
Comment (blancharda) [inline bundles/uds-core-swf/uds-bundle.yaml:710]: duplicate line fails the schema check

File: tasks/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+variables:
+  - name: ARCH
+    description: "What architecture to use"
+    default: "amd64"
+  - name: DEPLOY_IMAGE
+    description: "Container image to use to run uds deploy in"
+    default: "ghcr.io/defenseunicorns/build-harness/build-harness:2.0.28"
+  - name: BUNDLE_VERSION
+    description: "Version of the bundle to deploy"
+    default: "0.2.16"
```
Comment (anthonywendt) [inline tasks/deploy.yaml:10]:
```suggestion
    # x-release-please-start-version
    default: "0.2.16"
    # x-release-please-end
```

Do we need to add release please versioning to this default? If so we will also need to add this file to release-please-config.json in this project

File: bundles/uds-core-swf/uds-bundle.yaml [hunk 3]
```diff
@@ -50,6 +74,23 @@ packages:
     overrides:
       grafana:
         grafana:
+          # Remove if extra_volumes works
```
Comment (anthonywendt) [inline bundles/uds-core-swf/uds-bundle.yaml:77]: Remove?

File: bundles/uds-core-swf/uds-bundle.yaml [hunk 4]
```diff
@@ -179,6 +220,13 @@ packages:
                     includedNamespaces:
                       - gitlab
                     ttl: "240h"
+            - path: extraVolumes
+              value: *extra-volumes
+            - path: extraVolumeMounts
+              value: *extra-volume-mounts
+          # TODO: remove if not needed. Trying to updated trusted certs via container host first
```
Comment (anthonywendt) [inline bundles/uds-core-swf/uds-bundle.yaml:227]: Remove?

File: bundles/uds-core-swf/uds-bundle.yaml [hunk 5]
```diff
@@ -254,6 +322,11 @@ packages:
             - name: LOKI_S3_SECRET_ACCESS_KEY
               path: loki.storage.s3.secretAccessKey
               description: "The S3 Secret Access Key"
+            # Remove if extra_volumes works and S3 CA doesn't explicitly need set
```
Comment (anthonywendt) [inline bundles/uds-core-swf/uds-bundle.yaml:325]: Remove?

File: packages/additional-manifests/zarf.yaml [hunk 6]
```diff
@@ -33,9 +30,9 @@ components:
           - pepr-policy-exemptions/nutanix-csi-exemptions.yaml
           - pepr-policy-exemptions/metallb-exemptions.yaml
           - pepr-policy-exemptions/gitlab-exemptions.yaml
-  - name: mattermost-ca-secret
-    required: true
-    manifests:
-      - name: mattermost-ca-secret
-        files:
-          - mattermost/ca-secret.yaml
+  # - name: mattermost-ca-secret
```
Comment (anthonywendt) [inline packages/additional-manifests/zarf.yaml:33]: Is this not needed anymore? Remove?

**Comment by blancharda** [comment]
We should investigate providing the custom CA to neuvector.

The chart doesn't appear to support extravolume/mounts -- but maybe there's another way to provide it.
A cursory look suggests maybe something would be possible by setting `autoGenerateCert: false` and then manually providing the TLS cert/ca to the individual components (controller, monitor, scanner, adapter and enforcer)

**Comment by blancharda** [comment]
Same goes for Velero^

File: tasks/deploy.yaml (hunk 2 above)
Comment (blancharda) [reply tasks/deploy.yaml:10]: I think this is still a little WIP (?)
But I added a TODO comment just in case..

**Review by blancharda — DISMISSED** [review]
A few things to address in subsequent (or upstream) PRs (namely velero and neuvector) -- but this is a good start!

If we cut a release before addressing those^, we'll want to note it in the release.

**Review by blancharda — APPROVED** [review]
:shipit:

**Pull request merged**